src/Microsoft.ML.Data/DataView/AppendRowsDataView.cs

```diff
@@ -187,18 +187,21 @@ protected Delegate CreateGetter(int col)
 
             protected abstract ValueGetter<TValue> CreateTypedGetter<TValue>(int col);
 
-            public sealed override ValueGetter<TValue> GetGetter<TValue>(int col)
+            public sealed override ValueGetter<TValue> GetGetter<TValue>(int columnIndex)
             {
-                Ch.Check(IsColumnActive(col), "The column must be active against the defined predicate.");
-                if (!(Getters[col] is ValueGetter<TValue>))
+                Ch.Check(IsColumnActive(columnIndex), "The column must be active against the defined predicate.");
+                if (!(Getters[columnIndex] is ValueGetter<TValue>))
```

Comment:
` Ch.CheckParam(IsColumnActive(columnIndex), nameof(columnIndex), "requested column not active");`
from transposeLoader.cs
I like it more than some mythical `defined predicate`